src/core/Iterable.pm6

```diff
@@ -110,19 +110,29 @@ my role Iterable {
     }
 
     method hyper(Int(Cool) :$batch = 64, Int(Cool) :$degree = 4) {
+#?if !js
         HyperSeq.new:
           configuration =>
             HyperConfiguration.new(:$degree, :$batch, :method<hyper>),
           work-stage-head =>
             Rakudo::Internals::HyperIteratorBatcher.new(:$.iterator)
+#?endif
+#?if js
+        self
+#?endif
     }
 
     method race(Int(Cool) :$batch = 64, Int(Cool) :$degree = 4) {
+#?if !js
         RaceSeq.new:
           configuration =>
             HyperConfiguration.new(:$degree, :$batch, :method<race>),
           work-stage-head =>
             Rakudo::Internals::HyperIteratorBatcher.new(:$.iterator)
+#?endif
+#?if js
+        self
```

Comment:
This is probably technically ok since the return type isn't constrained, but I feel like it should be ( and thus the `#!js` logic would end up in `RaceSeq`/`HyperSeq` ) due to stuff like `isa-ok (^1000).hyper.map(*+1).race, RaceSeq, 'Can switch from hyper to race mode';` ( from the roast ).

Reply:
Done. I'm returning RaceSeq/HyperSeq which are Seq subclasses on the js backend now.